libsplinter/src/auth/oauth/builder/openid.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use reqwest::blocking::Client;
+
+use crate::auth::oauth::{builder::OAuthClientBuilder, error::OAuthClientBuildError, OAuthClient};
+use crate::auth::rest_api::identity::{openid::OpenIdUserIdentityProvider, IdentityProvider};
+use crate::error::{InternalError, InvalidStateError};
+
+/// Builds a new `OAuthClient` using an OpenID discovery document.
+pub struct OpenIdOAuthClientBuilder {
+    openid_discovery_url: Option<String>,
+    inner: OAuthClientBuilder,
+}
+
+impl OpenIdOAuthClientBuilder {
+    /// Constructs a new [`OpenIdOAuthClientBuilder`].
+    pub fn new() -> Self {
+        Self {
+            openid_discovery_url: None,
+            inner: OAuthClientBuilder {
```

Comment:
Explicitly using ::new() here is both less code and avoids bypassing the constructor.

Reply:
fixed